external/error_report/error_report.c

```diff
@@ -0,0 +1,205 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <libgen.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <sched.h>
+#include <error_report/error_report.h>
+#include "sched/sched.h"
+
+#define ERR_IP_ADRR_LEN 16
+
+#ifdef CONFIG_ERROR_REPORT_DEBUG
+#define rt_errdebug dbg
```

Comment:
Please don't use dbg directly. You should define this module specifc dbg like ndbg, sdbg.

Reply:
Will address this in the following change to commit.